AppKit/CPTextView/CPTextView.j

```diff
@@ -899,35 +900,74 @@ var kDelegateRespondsTo_textShouldBeginEditing
                           }, 500);
 }
 
+- (CGPoint)_characterIndexFromRawPoint:(CGPoint)point
+{
+    var fraction = [],
+        point = [self convertPoint:point fromView:nil];
+    
+    // convert to container coordinate
+    point.x -= _textContainerOrigin.x;
+    point.y -= _textContainerOrigin.y;
+    
+    var index = [_layoutManager glyphIndexForPoint:point inTextContainer:_textContainer fractionOfDistanceThroughGlyph:fraction];
+    
+    if (index === CPNotFound)
+        index = [_layoutManager numberOfCharacters];
+    else if (fraction[0] > 0.5)
+        index++;
+    
+    return index;
+}
+- (CGPoint)_characterIndexFromEvent:(CPEvent)event
+{
+    return [self _characterIndexFromRawPoint:[event locationInWindow]];
+}
 
 #pragma mark -
 #pragma mark Mouse Events
 
 - (void)mouseDown:(CPEvent)event
 {
     [_CPNativeInputManager cancelCurrentInputSessionIfNeeded];
-
-    var fraction = [],
-        point = [self convertPoint:[event locationInWindow] fromView:nil],
-        granularities = [CPNotFound, CPSelectByCharacter, CPSelectByWord, CPSelectByParagraph];
-
     [_caret setVisibility:NO];
-
-    // convert to container coordinate
-    point.x -= _textContainerOrigin.x;
-    point.y -= _textContainerOrigin.y;
-
-    _startTrackingLocation = [_layoutManager glyphIndexForPoint:point inTextContainer:_textContainer fractionOfDistanceThroughGlyph:fraction];
-
-    if (_startTrackingLocation === CPNotFound)
-        _startTrackingLocation = [_layoutManager numberOfCharacters];
-    else if (fraction[0] > 0.5)
-        _startTrackingLocation++;
-
+    
+    _startTrackingLocation = [self _characterIndexFromEvent:event];
+    
+    var granularities = [CPNotFound, CPSelectByCharacter, CPSelectByWord, CPSelectByParagraph];
     [self setSelectionGranularity:granularities[[event clickCount]]];
-
+    
+    if ([self selectionGranularity] == CPSelectByCharacter && CPLocationInRange(_startTrackingLocation, _selectionRange))
+    {
+        _movingSelection = CPMakeRange(_startTrackingLocation, 0);
+        var placeholderRect = CGRectIntersection([_layoutManager boundingRectForGlyphRange:_selectionRange inTextContainer:_textContainer], _frame),
+            dragPlaceholder;
+        
+        placeholderRect.size.width += 10; // prevent wrapping
+        dragPlaceholder = [[CPTextView alloc] initWithFrame:placeholderRect];
```

Comment:
I see 2 possible optimizations here:
1/ Make dragPlaceholder a lazy stored instance variable
2/ Since dragPlaceholder is a private temporary or reusable view and nobody interacts with it, why not just use  a div or span element instead of the full CPTextView. It would be much lightweight to create and move. This is if the perf is important of course.